### PR TITLE
Return Hardsuits To Lockers

### LINKED ISSUE #16431: warden has 2 hardsuits
## Description
1 in locker 1 in suit storage

i will fix in a bit if nobody else does

### PR DESCRIPTION
**Pull request by Nairodian**
## About the PR
Adds a new fill prototype for all lockers, so mappers may choose if they want suit storage units or locker bound hardsuits.
**The default fill will have no hardsuit.** So maps that haven't been updated yet with either a storage unit or the new type of locker will lack hardsuits.

Makes #16433 obsolete.
Fixes #16431

**Media**
- [X] This PR does not require an ingame showcase

**Changelog**
No changelog needed due to it being mapper heavy content.

### CODE REVIEW

**Review by metalgearsloth — APPROVED** [review]

**Comment by Emisse** [comment]
better solution would be making a 2nd prototype for lockers than has hardsuits included i thinks, that way theres more options when mapping

**Comment by deltanedas** [comment]
would be nice for small maps to have locker fills that have suits say for warden on saltern which is really cramped

**Comment by Nairodian** [comment]
Sure, what suffix should it have? Can't really come up with any short good sounding ones off the top of my head.

**Comment by Emisse** [comment]
[filled] [hardsuit]

**Comment by Nairodian** [comment]
Aight, done and done. PR description changed to fit.

**Comment by 0x6273** [comment]
> **The default fill will have no hardsuit**

Why? Can't you just make the new prototype have no hardsuit instead of removing them from the existing prototype, to avoid breaking all maps?

File: Resources/Prototypes/Catalog/Fills/Lockers/cargo.yml [hunk 1]
```diff
@@ -10,4 +29,4 @@
       - id: ClothingBeltUtilityFilled
         prob: 0.8
       - id: SurvivalKnife
-      - id: WeaponProtoKineticAccelerator
+      - id: WeaponProtoKineticAccelerator
```
Comment (0x6273) [inline Resources/Prototypes/Catalog/Fills/Lockers/cargo.yml:32]: You need to fix your editor so it saves with proper linebreaks.

File: Resources/Prototypes/Catalog/Fills/Lockers/engineer.yml [hunk 2]
```diff
@@ -89,10 +89,24 @@
       - id: GasAnalyzer
       - id: MedkitOxygenFilled
       - id: HolofanProjector
-
+
```
Comment (0x6273) [inline Resources/Prototypes/Catalog/Fills/Lockers/engineer.yml:92]: Trailing whitespace. Fix your editor.

File: Resources/Prototypes/Catalog/Fills/Lockers/engineer.yml [hunk 3]
```diff
@@ -104,6 +118,18 @@
...
+  suffix: Filled
+  parent: LockerEngineer
+  components:
+  - type: StorageFill
+    contents:
+      - id: ClothingHandsGlovesColorYellow
+      - id: ClothingMaskGas
+      - id: trayScanner
+
+
```
Comment (0x6273) [inline Resources/Prototypes/Catalog/Fills/Lockers/engineer.yml:132]:
```suggestion

```
Only one newline between prototypes

**Review by 0x6273 — CHANGES_REQUESTED** [review]

**Comment by Nairodian** [comment]
> Why? Can't you just make the new prototype have no hardsuit instead of removing them from the existing prototype, to avoid breaking all maps?

Firstly, because that was Emisse's direct wording. Secondly, _most_ maps have been adapted to have suit storage units now, meaning it's less work over all and won't double the suit count.

**Comment by 0x6273** [comment]
> > Why? Can't you just make the new prototype have no hardsuit instead of removing them from the existing prototype, to avoid breaking all maps?
> 
> Firstly, because that was Emisse's direct wording. Secondly, _most_ maps have been adapted to have suit storage units now, meaning it's less work over all and won't double the suit count.

Ah, I wasn't aware maps had been updated. I still think they shouldn't have been removed from existing lockers, but if most maps are updated it makes sense to do it this way.

**Pull request merged**